src/components/ListBoxField/ListBoxField.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ListBoxLabel from '../ListBoxLabel';
+import ListBoxMenuIcon from '../ListBoxMenuIcon';
+
+const ListBoxField = ({
+  type = 'default',
+  label,
+  selectedItem,
+  clearSelection,
+  isOpen,
+  // `rest` here should correspond with `getButtonProps` from `downshift`
+  ...rest
+}) => {
+  const className = cx({
+    'bx--list-box__field': true,
+    'bx--list-box__field--inline': type === 'inline',
+    'bx--list-box__field--open': isOpen,
```

Comment:
Great to consolidate `--open` state class to the top-level list box component. In this way, we don't need to have multiple component managing `--open` state class.